spec.html

```diff
@@ -17142,9 +17142,9 @@ <h2>Syntax</h2>
     <emu-grammar type="definition">
       IfStatement[Yield, Await, Return] :
         `if` `(` Expression[+In, ?Yield, ?Await] `)` Statement[?Yield, ?Await, ?Return] `else` Statement[?Yield, ?Await, ?Return]
-        `if` `(` Expression[+In, ?Yield, ?Await] `)` Statement[?Yield, ?Await, ?Return]
+        `if` `(` Expression[+In, ?Yield, ?Await] `)` Statement[?Yield, ?Await, ?Return] [lookahead &ne; `else`]
     </emu-grammar>
-    <p>Each `else` for which the choice of associated `if` is ambiguous shall be associated with the nearest possible `if` that would otherwise have no corresponding `else`.</p>
+    <p>Note that the &ldquo;[lookahead &ne; `else`]&rdquo; has the effect of always associating an `else` to the nearest `if`, by only allowing `if`-`else` inside the truthy branch of an `if`-`else`.</p>
```

Comment:
re "the nearest `if`": Well, not really. The *nearest* `if` might actually be after the `else`. So you could say "the nearest *preceding* `if`", but even that's not correct, because that might be within the truthy arm of the statement. You're probably better off describing it more like the existing text does.

Reply:
re "by only allowing `if`-`else` inside the truthy branch of an `if`-`else`." This phrasing is very ambiguous. (I can think of 3 interpretations.) It would be clearer to say "by disallowing an else-less `if` as the truthy branch of an `if`-`else`. (But I'm not sure the phrase helps anyway.)

Reply:
Fair enough about ambiguity. I'm fine with converting the current normative prose into a note that says the effect of

"[lookahead &ne; else] is has the effect of associating each `else` for which the choice of associated `if` with the nearest possible `if` that would otherwise have no corresponding `else`".

Thoughts about that, @jmdyck?

Reply:
First, I agree it should be a Note. As for the wording, you'd need to delete the "is", and insert a predicate after the first "if", something like:

> [lookahead ≠ else] has the effect of associating each `else` for which the choice of associated `if` would otherwise be ambiguous with the nearest possible `if` that would otherwise have no corresponding `else`.

That's probably accurate enough for a Note, but I think the subordinate clauses make it a bit hard to read. I'd suggest something like:

> The lookahead-restriction [lookahead ≠ else] resolves the classic "dangling else" problem in the usual way. That is, when the choice of associated `if` is otherwise ambiguous, the `else` is associated with the nearest (innermost) of the candidate `if`s.

Reply:
```suggestion
    <p>Note that the <emu-grammar>[lookahead &ne; `else`]</emu-grammar> has the effect of associating each `else` for which the choice of associated `if` is ambiguous with the nearest possible `if` that would otherwise have no corresponding `else`.</p>
```

Reply:
I'm pretty sure the use of `<emu-grammar>` here is incorrect -- everywhere else in the spec, the content of an `<emu-grammar>` element is the text for one or more productions.

Also, "for which the choice is ambiguous" is incorrect -- with the lookahead-restriction, there is no ambiguity. Instead, what's of of interest is the ambiguity that *would* exist if the lookahead-restriction *weren't* there. 

And I think the words "is ambiguous with" are a garden path.

Reply:
When given a choice between having `<emu-grammar>` contain grammar portions that are not full productions and having portions of grammar that are not marked up at all, I would prefer the former. I don't see any reason to avoid it.

@jmdyck Do you have a concrete suggestion for your preferred wording?

Reply:
> When given a choice between having `<emu-grammar>` contain grammar portions that are not full productions and having portions of grammar that are not marked up at all, I would prefer the former. I don't see any reason to avoid it.

FWIW in other places we use `<sub>`, as in

```html
<emu-grammar>
  BindingIdentifier[Yield, Await] : `yield`
</emu-grammar>
<ul>
  <li>
    It is a Syntax Error if this production has a <sub>[Yield]</sub> parameter.
  </li>
</ul>
```

Reply:
> When given a choice between having `<emu-grammar>` contain grammar portions that are not full productions and having portions of grammar that are not marked up at all, I would prefer the former.

I almost completely disagree. But note that it's not either-or: we can reserve `<emu-grammar>` for complete productions *and* have other grammar fragments marked up in some other way. E.g., ecmarkup renders lookahead-restrictions in [`<emu-gann>`](https://tc39.es/ecmarkup/#emu-gann) elements.

> I don't see any reason to avoid it.

One reason: software that reads the spec would have to be prepared for a larger set of possibilities for the expected content of the element.

Another reason: I don't think grammarkdown can handle it. 

> @jmdyck Do you have a concrete suggestion for your preferred wording?

Yup, I suggested it [2 days ago](https://github.com/tc39/ecma262/pull/1966#discussion_r424804607).